File: javascript_small_problems/medium_problems2/unlucky_days.js
function fridayThe13ths(year) {
  let count = 0;
  for (let month = 0; month < 12; month += 1) {
    let day = new Date(year, month, 13).getDay();
    if (day === 5) count += 1;
  }

  return count;
}

console.log(fridayThe13ths(1986));      // 1
console.log(fridayThe13ths(2015));      // 3
console.log(fridayThe13ths(2017));      // 2

File: small_problems/medium2/unlucky_days.rb
def friday_13th(year)
  result = 0
  month = 0

  12.times do
    month += 1
    thirteenth_date = Time.local(year, month, 13)
    result += 1 if thirteenth_date.friday?
  end

  result
end

# alternate

require 'date'

def friday_13th(year)
  unlucky_count = 0
  thirteenth = Date.civil(year, 1, 13)
  12.times do
    unlucky_count += 1 if thirteenth.friday?
    thirteenth = thirteenth.next_month
  end
  unlucky_count
end

p friday_13th(2015) == 3
p friday_13th(1986) == 1
p friday_13th(2019) == 2
